### DISCUSSION TITLE
Active broadcast

### DISCUSSION THREAD

**Discussion by dodeyeikoi** [Cyfrin/foundry-full-course-cu#1097] [Q&A]
When I run ```forge test``` on my RaffleTest.t.sol contract, I get this error:
````
cryptoace@Dodeye:~/solidity-course/foundry-f23/foundry-smart-contract-lottery-f23$ forge test --mt testCantEnterWhenRaffleIsCalculating
[⠔] Compiling...
[⠔] Compiling 3 files with 0.8.19
[⠑] Solc 0.8.19 finished in 3.98s
Compiler run successful!
2023-11-09T06:13:02.388833Z ERROR forge::runner: setUp failed reason="You have an active broadcast already." contract=0x34a1d3fff3958843c43ad80f30b94c510645c316

Running 1 test for test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: Setup failed: You have an active broadcast already.] setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 12.58ms
 
Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: Setup failed: You have an active broadcast already.] setUp() (gas: 0)

Encountered a total of 1 failing tests, 0 tests succeeded
````

this is my RaffleTest.t.sol contract:
````
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {DeployRaffle} from "../../script/DeployRaffle.s.sol";
import {Raffle} from "../../src/Raffle.sol";
import {Test, console} from "forge-std/Test.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";

contract RaffleTest is Test {
    /* Events */
    event EnteredRaffle(address indexed player);
    
    Raffle raffle;
    HelperConfig helperConfig;

    uint256 entranceFee;
    uint256 interval;
    address vrfCoordinator;
    bytes32 gasLane; // keyHash
    uint64 subscriptionId;
    uint32 callbackGasLimit;
    address link;

    address public PLAYER = makeAddr("player");
    uint256 public constant STARTING_USER_BALANCE = 10 ether;

    function setUp() external {
        DeployRaffle deployer = new DeployRaffle();
        (raffle, helperConfig) = deployer.run();
        (
            entranceFee,
            interval,
            vrfCoordinator,
            gasLane, // keyHash
            subscriptionId,
            callbackGasLimit,
            link
        ) = helperConfig.activeNetworkConfig();
        vm.deal(PLAYER, STARTING_USER_BALANCE);
    }

    function testRaffleInitializesInOpenState() public view {
        assert(raffle.getRaffleState() == Raffle.RaffleState.OPEN);
    }

    //////////////////////////
    ////// ENTER RAFFLE //////
    //////////////////////////

    function testRaffleRevertsWhenYouDontPayEnough() public {
        // Arrange
        vm.prank(PLAYER);
        // Act // Assert
        vm.expectRevert(Raffle.Raffle__Not_Enough_ETH_Sent.selector);
        raffle.enterRaffle();
    }

    function testRaffleRecordsPlayerWhenTheyEnter() public {
        vm.prank(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
        address playerRecorded = raffle.getPlayer(0);
        assert(playerRecorded == PLAYER);
    }

    function testEmitsEventOnEntrance() public {
        vm.prank(PLAYER);
        vm.expectEmit(true, false, false, false, address(raffle));
        emit EnteredRaffle(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
    }

    function testCantEnterWhenRaffleIsCalculating() public {
        vm.prank(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
        vm.warp(block.timestamp + interval + 1);
        vm.roll(block.number + 1);
        raffle.performUpkeep("");

        vm.expectRevert(Raffle.Raffle__RaffleNotOpen.selector);
        vm.prank(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
    }
}
````

what am I doing wrong please?

**Comment by Abhishek-2416** [comment]
You are starting a broadcast before stopping the already running one. Check your setUp function if it's conflicting with any other running broadcast

**Reply by dodeyeikoi** [reply]
I have checked and I don't see any running broadcast conflicting, I uploaded the raffletest contract above, please check if you see any

**Comment by dodeyeikoi** [comment]
I need help here please, I haven't solved this yet
@PatrickAlphaC

**Reply by Abhishek-2416** [reply]
Did you find a solution to this ?

**Comment by Merulez** [comment]
Damn , i have the exact same issue and i cant resolve it

**Reply by Abhishek-2416** [reply]
@PatrickAlphaC  Please help us with this

**Reply by Merulez** [reply]
If i remove the vm.startBroadcast() and the vm.stopBroadcast() from the DeployRaffle.s.sol contract it passes. 
but i have to remove also the vm.prank(PLAYER) and now i am confused :D

**Reply by 0xusmanf** [reply]
Hello guys, can you please share the link to your project repo?

**Comment by Merulez** [comment]
Yo broo @dodeyeikoi  i found what we were missing , so go to HelperConfig.s.sol and you have to type vm.stopBroadcast() 
This is the code :
 function getOrCreateAnvilEthConfig() public returns(NetworkConfig memory) {
        if(activeNetworkConfig.vrfCoordinator != address(0)) {
            return activeNetworkConfig;
        }

        uint96 baseFee = 0.25 ether;
        uint96 gasPriceLink = 1e9;
        vm.startBroadcast();
        VRFCoordinatorV2Mock vrfCoordinatorMock = new VRFCoordinatorV2Mock(
            baseFee,
            gasPriceLink
        );
        vm.stopBroadcast();
        return NetworkConfig({
            entranceFee: 0.01 ether,
            interval: 30 seconds,
            vrfCoordinator: address(vrfCoordinatorMock),
            gasLane: 0x474e34a077df58807dbe9c96d3c009b23b3c6d0cce433e59bbf5b34f823bc56c,
            subscriptionId: 0, // update this with our subId
            callbackGasLimit: 500000
        });
    }

**Comment by thomasender** [comment]
For future readers, I just also had this issue and my error was in `HelperConfig.s.sol` I wrote this:
```

        vm.startBroadcast();
        // Deploy a new VRFCoordinatorV2Mock
        VRFCoordinatorV2Mock vrfCoordinatorMock = new VRFCoordinatorV2Mock(
            baseFee,
            gasPriceLink
        );
       vm.startBroadcast();
 ```
 
instead of this:
```

        vm.startBroadcast();
        // Deploy a new VRFCoordinatorV2Mock
        VRFCoordinatorV2Mock vrfCoordinatorMock = new VRFCoordinatorV2Mock(
            baseFee,
            gasPriceLink
        );
        vm.stopBroadcast();
        
```

**Comment by TechTunist** [comment]
I had this issue because in DeployRaffle.s.sol, I had put the vm.stopBroadcast() after the AddConsumer line after deploying the raffle contract. Check the code within the start / stopBroadcast calls does not contain unwanted elements

**Comment by wharika18** [comment]
In my case , I missed appending parentheses to the end of vm.stopBroadcast in HelperConfig.s.sol. Meaning , i typed mere vm.stopBroadcast instead of vm.stopBroadcast();

**Comment by DCVglobalnetwork** [comment]
in my case I deleted and compile and than in order to deploy on the local blockchain I had to add to the code again only than I was able to accomplish the project

**Comment by xyl2004** [comment]
i think you just written startBroadcast in another file and didn't write stopBroadcast

**Comment by EngrPips** [comment]
Interesting.
